src/data/reports/mal-data-approval/UserGroupD2Repository.ts

```diff
@@ -0,0 +1,20 @@
+import { D2Api } from "@eyeseetea/d2-api/2.34";
+import { UserGroupRepository } from "../../../domain/reports/mal-data-approval/repositories/UserGroupRepository";
+import { Ref } from "../../../domain/common/entities/Base";
+
+export class UserGroupD2Repository implements UserGroupRepository {
+    constructor(private api: D2Api) {}
+
+    async getUserGroupByCode(code: string): Promise<Ref> {
```

Comment:
Not a big deal, but let's be more explicit here and use the name of the entity (UserGroup) just to avoid any possible confusion.